compiler/shared_ast/expr.ml

```diff
@@ -536,11 +536,7 @@ and equal : type a. (a, 't) gexpr -> (a, 't) gexpr -> bool =
   | EArray es1, EArray es2 -> equal_list es1 es2
   | ELit l1, ELit l2 -> l1 = l2
   | EAbs { binder = b1; tys = tys1 }, EAbs { binder = b2; tys = tys2 } ->
-    Type.equal_list tys1 tys2
-    &&
-    let vars1, body1 = Bindlib.unmbind b1 in
-    let body2 = Bindlib.msubst b2 (Array.map (fun x -> EVar x) vars1) in
-    equal body1 body2
+    Type.equal_list tys1 tys2 && Bindlib.eq_mbinder equal b1 b2
```

Comment:
Good catch

Reply:
Well the previous version was OK, but I had'nt discovered this function from `Bindlib` yet :)